src/linear_algebra.jl

```diff
@@ -27,16 +27,16 @@ function mutable_operate_to!(C::Vector, ::typeof(*), A::AbstractMatrix, B::Abstr
     end
 
     # We need a buffer to hold the intermediate multiplication.
-    mul_buffer = zero(zero(eltype(A)) * zero(eltype(B)))
-    for k = 1:mB
-        aoffs = (k-1)*Astride
+    mul_buffer = zero(promote_operation(*, eltype(A), eltype(B)))
+    @inbounds for k = Base.OneTo(mB)
```

Comment:
Adding another `@inbounds` is redundant here: https://stackoverflow.com/questions/38901275/inbounds-propagation-rules-in-julia

Reply:
Indeed, thanks for catching this